web-ui/src/pages/KudosHomePage.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+
+import { UPDATE_TOAST } from '../context/actions';
+import { AppContext } from '../context/AppContext';
+import { selectCsrfToken } from '../context/selectors';
+import { sortKudos } from '../context/util';
+
+import { getAllKudos } from '../api/kudos';
+
+import KudosCard from '../components/kudos_card/KudosCard';
+import KudosDialog from '../components/kudos_dialog/KudosDialog';
+import SkeletonLoader from '../components/skeleton_loader/SkeletonLoader';
+
+import StarIcon from '@mui/icons-material/Star';
+import { Button, Grid, Typography } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import './KudosPage.css';
+
+const PREFIX = 'KudosPage';
+const classes = {
+  expandOpen: `${PREFIX}-expandOpen`,
+  expandClose: `${PREFIX}-expandClose`
+};
+
+const Root = styled('div')({
+  [`& .${classes.expandOpen}`]: {
+    transform: 'rotate(180deg)',
+    transition: 'transform 0.1s linear',
+    marginLeft: 'auto'
+  },
+  [`& .${classes.expandClose}`]: {
+    transform: 'rotate(0deg)',
+    transition: 'transform 0.1s linear',
+    marginLeft: 'auto'
+  }
+});
+
+const KudosHomePage = () => {
+  const { state, dispatch } = useContext(AppContext);
+  const csrf = selectCsrfToken(state);
+
+  const [kudos, setKudos] = useState(null);
+  const [kudosDialogOpen, setKudosDialogOpen] = useState(false);
+  const [kudosLoading, setKudosLoading] = useState(false);
+
+  let lastMonth = new Date();
+  lastMonth.setMonth(lastMonth.getMonth() - 1);
+
+  const loadKudos = useCallback(async () => {
+    setKudosLoading(true);
+    const res = await getAllKudos(csrf, true);
+    if (res.error) {
```

Comment:
I think this whole if block should be replaced by this:

```js
if (res.error) return;
```

I say this because `getAllKudos` calls the `resolve` function and that takes care of displaying a toast message if the endpoint returns an error.